k9mail/src/main/java/com/fsck/k9/fragment/MessageListAdapter.java

```diff
@@ -320,20 +344,37 @@ private Drawable buildStatusHolder(boolean forwarded, boolean answered) {
         return null;
     }
 
-    private void setBackgroundColor(View view, boolean selected, boolean read) {
+    private void setBackgroundColor(
+        View view, boolean selected, boolean read, Keyword firstTag)
+    {
         if (selected || K9.useBackgroundAsUnreadIndicator()) {
             int res;
             if (selected) {
                 res = R.attr.messageListSelectedBackgroundColor;
-            } else if (read) {
-                res = R.attr.messageListReadItemBackgroundColor;
             } else {
-                res = R.attr.messageListUnreadItemBackgroundColor;
+                if (read) {
+                    res = R.attr.messageListReadItemBackgroundColor;
+                } else {
+                    res = R.attr.messageListUnreadItemBackgroundColor;
+                }
             }
 
             TypedValue outValue = new TypedValue();
             fragment.getActivity().getTheme().resolveAttribute(res, outValue, true);
-            view.setBackgroundColor(outValue.data);
+            int backgroundColor = outValue.data;
+
```

Comment:
Move all this to a helper method.

Reply:
Created `MessageListAdapter.blendBackgroundWithKeyword()`.